### PR TITLE
Update podcasts grid design

### PR DESCRIPTION
**Pull request by ashiagr**
Part of #2081 

## Description

This updates Podcasts tab grid design.
Figma Design: G60UqxPhTynbalo0Vfr1NC-fi-1255_8864

** I created a separate feature flag for these changes currently disabled in Firebase. It can be enabled in 7.64 if required.

## Testing Instructions
1. Open Podcasts tab
2. Notice the grid spacing matches the design
3. Tap more menu in the toolbar
4. Select All unfinished for Badges
5. ✅ Notice that a new unfinished badge is shown
6. Select Only latest episode
7. ✅ Notice that a new only latest badge is shown
8. Switch to list view
9. ✅ Notice that new badges(similar to iOS) are shown for the list view ** <- (Needs clarification G60UqxPhTynbalo0Vfr1NC-fi-1255_8864#793222421.  I matched it with iOS.)
10. Disable "Podcast grid view design changes" feature flag 
11. Kill and Restart the app
12. ✅ Notice that original grid design changes are shown

## Screenshots or Screencast 

out-1

out-2

## Checklist
- [x] If this is a user-facing change, I have added an entry in CHANGELOG.md
- [x] Ensure the linter passes (`./gradlew spotlessApply` to automatically apply formatting/linting)
- [ ] I have considered whether it makes sense to add tests for my changes N/A
- [ ] All strings that need to be localized are in `modules/services/localization/src/main/res/values/strings.xml` N/A
- [x] Any jetpack compose components I added or changed are covered by compose previews
- [ ] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics. N/A
 
#### I have tested any UI changes...

- [x] with different themes
- [x] with a landscape orientation
- [x] with the device set to have a large display and font size
- [ ] for accessibility with TalkBack

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add tests for class <code>CountBadgeStyle</code> (or add <code>unit-tests-exemption</code> label to ignore this).</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by mebarbosa — COMMENTED** [review]
It looks good to me.

The only difference is the badge for `only latest badge is shown` if we compare with iOS, but since we are already waiting for clarification we can address this later

**Review by mebarbosa — APPROVED** [review]

**Comment by ashiagr** [comment]
I added a shadow for podcast grid items in [f660207](https://github.com/Automattic/pocket-casts-android/pull/2165/commits/f66020775ba41005fe6877665c578bc054d245a7) (Ref: https://github.com/Automattic/pocket-casts-ios/issues/1690#issue-2275374478) and increased small badge size from 10dp to 12dp (Ref: https://github.com/Automattic/pocket-casts-ios/issues/1689).  

Shadow


Badge Size
 



Can you take another look @mebarbosa?

File: modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/podcasts/FolderAdapter.kt [hunk 1]
```diff
@@ -178,12 +179,15 @@ class FolderAdapter(
...
         val podcastThumbnail: ImageView = view.findViewById(R.id.podcast_artwork)
+        val podcastCardView = view.findViewById<CardView>(R.id.podcast_card_view)
         val podcastBackground: View? = view.findViewById(R.id.header_background)
         val podcastTitle: TextView = view.findViewById(R.id.library_podcast_title)
         val author: TextView? = view.findViewById(R.id.podcast_author)
         val unplayedText: TextView = view.findViewById(R.id.unplayed_count)
         val unplayedBackground: ImageView? = view.findViewById(R.id.unplayed_background)
         val countTextMarginSmall: Int = 2.dpToPx(view.resources.displayMetrics)
+        val cardElevation: Float = 2.dpToPx(view.resources.displayMetrics).toFloat()
+        val cardCornerRadius: Float = 4.dpToPx(view.resources.displayMetrics).toFloat()
```
Comment (mebarbosa) [inline modules/features/podcasts/src/main/java/au/com/shiftyjelly/pocketcasts/podcasts/view/podcasts/FolderAdapter.kt:190]: Really small suggestion: What do you think we add a top padding for the badge? 

I feel the new drop-shadow is clipping a bit the podcast above. See:

Screenshot_20240508_093732_Pocket Debug~2

and see the proposal [here](https://github.com/Automattic/pocket-casts-ios/issues/1690)

**Comment by MiSikora** [comment]
The badge counter does not respond to font size. Is this intended?

Screenshot_20240508-152438

**Comment by ashiagr** [comment]
@mebarbosa 👋 

I've reduced badge offset in [343cac4](https://github.com/Automattic/pocket-casts-android/pull/2165/commits/343cac4b28a8181c9ff986e5f120a13e17f4dd5a)



> The badge counter does not respond to font size. Is this intended?

It is intentional as on a large display, the grid items remain the same size but the badge will cover the artwork too much.

**Pull request merged**